### PR TITLE
1.0.0.rc4

### PR DESCRIPTION
**Pull request by mamoonraja**
This PR:

- update examples to use `Authenticators`
- refactor `Authenticators` and move them into their own module
- add support for Visual Recognition V4

### CODE REVIEW

File: test/integration/test_visual_recognition_v4.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            "data": image_file_1,
+            "filename": "face.jpg",
+            "content_type": "image/jpeg"
+          }
+
+        ],
+        collection_ids: @collection_id,
+        features: "objects"
+      ).result
+      refute(result.nil?)
```
Comment (mediumTaj) [inline test/integration/test_visual_recognition_v4.rb]: Can we check here that the result had two images analyzed?

Comment (mamoonraja) [reply test/integration/test_visual_recognition_v4.rb]: It doesn't, I am working on it on the core side.

Comment (mamoonraja) [reply test/integration/test_visual_recognition_v4.rb]: fix added: https://github.com/watson-developer-cloud/ruby-sdk/pull/80/commits/ddc3cda10e7746976f81d161be8f8c417961faa8

**Comment by mediumTaj** [comment]
Mamoon can you delete visrec models and regenerate with api def `09a54edf1fa2f715752a8d9f6f7d942c3cb4d8b4`

File: examples/assistant_v1.rb [hunk 2]
```diff
@@ -1,21 +1,26 @@
 # frozen_string_literal: true
 
-require("ibm_watson/assistant_v1")
-require("json")
+require "json"
+require "ibm_watson/authenticators"
+require "ibm_watson/assistant_v1"
```
Comment (mediumTaj) [inline examples/assistant_v1.rb]: multi line comment!

File: lib/ibm_watson/visual_recognition_v4.rb [hunk 3]
```diff
@@ -0,0 +1,658 @@
...
+    # Analyze images.
+    # Analyze images by URL, by file, or both against your own collection. Make sure
+    #   that **training_status.objects.ready** is `true` for the feature before you use a
+    #   collection to analyze images.
+    #
+    #   Encode the image and .zip file names in UTF-8 if they contain non-ASCII
+    #   characters. The service assumes UTF-8 encoding if it encounters non-ASCII
+    #   characters.
+    # @param collection_ids [String] The IDs of the collections to analyze. Separate multiple values with commas.
+    # @param features [String] The features to analyze. Separate multiple values with commas.
```
Comment (mediumTaj) [inline lib/ibm_watson/visual_recognition_v4.rb]:
```
 # @param collection_ids [String] The IDs of the collections to analyze.
 # @param features [String] The features to analyze.
```
These should also be input as an array of strings

**Pull request merged**